lib/chef/resource/apt_package.rb

```diff
@@ -56,8 +56,8 @@ class AptPackage < Chef::Resource::Package
       **Prevent the apt_package resource from installing packages with pattern matching names**:
 
       By default, the apt_package resource will install the named package.
-      If it can't find a package with the exact same name, it will treat the package name as regex string and match with any package that matches that regex.
-      This may lead Chef Infra Client to install one or more packages with names that match that regex.
+      If it can't find a package with the exact same name, it will treat the package name as regular expression string and match with any package that matches that regular expression.
```

Comment:
@jaymzh also mentioned this https://github.com/chef/chef/pull/13873/files#r1348641802 ... basically, we don't want to imply that the unanchored expression is not a regex.